src/Emulator/Peripherals/Peripherals/MTD/OpenTitan_FlashController.cs

```diff
@@ -0,0 +1,1185 @@
+                DoEraseDataPage(workItem.erasePageAddress);
+            }else if(workItem.eraseInfoPage){
+                DoEraseInfoPage(workItem.erasePageAddress);
+            }else if(workItem.readData){
+                DoReadData(workItem.readAddress);
+            }else if(workItem.readInfo){
+                DoReadInfo(workItem.readAddress);
+            }else if(workItem.programData){
+                DoProgramData(workItem.programAddress);
+            }else if(workItem.programInfo){
+                DoProgramInfo(workItem.programAddress);
+            }else{
+                this.Log(
+                    LogLevel.Warning,
+                    "OpenTitan_FlashController/DoWork: Unhandled workItem");
+            }
+        }
+
+        private void DoEraseBank(uint bankNumber){
+            /*
+            ** TODO(julianmb): memory protection should be implemented.
+            ** It currently doesnt have a unit test though.
+            ** MP_BANK_CFG register is of interest
+            */
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/DoEraseBank: bankNumber = {0}",
+                bankNumber);
+
+            long bytesPerBank = FLASH_PAGES_PER_BANK * FLASH_WORDS_PER_PAGE * FLASH_WORD_SZ;
+            long bankEflashOffset = bankNumber * bytesPerBank;
+
+            for(long i = 0 ; i < bytesPerBank ; i++){
+                flash.WriteByte(bankEflashOffset + i, 0xff);
+            }
+        }
+
+        private void DoEraseDataPage(long pageAddress){
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/DoEraseDataPage: pageAddress = 0x{0:X}",
+                pageAddress);
+            if(IsOperationAllowed(OperationType.ERASE_DATA_PAGE, pageAddress)){
+                long pageOffset = pageAddress - FLASH_MEM_BASE_ADDR;
+                long bytesPerPage = FLASH_WORDS_PER_PAGE * FLASH_WORD_SZ;
+                for(long i = 0 ; i < bytesPerPage ; i++){
+                    flash.WriteByte(pageOffset + i, 0xff);
+                }
+            }else{
+                interruptStatusRegisterOpErrorFlag.Value |= true;
+                opStatusRegisterErrorFlag.Value |= true;
+            }
+        }
+
+        private void DoEraseInfoPage(long pageAddress){
+            /*
+            ** TODO(julianmb): memory protection should be implemented.
+            ** It currently doesnt have a unit test though.
+            ** BANK*_INFO*_PAGE_CFG_* registers are of interest
+            */
+            this.Log(
+                LogLevel.Noisy,
+                "OpenTitan_FlashController/DoEraseInfoPage: pageAddress = 0x{0:X}",
+                pageAddress);
+            long pageOffset = pageAddress - FLASH_MEM_BASE_ADDR;
+            long bytesPerPage = FLASH_WORDS_PER_PAGE * FLASH_WORD_SZ;
+            for(long i = 0 ; i < bytesPerPage ; i++){
+                infoFlash.WriteByte(pageOffset + i, 0xff);
+            }
```

Comment:
This looks like very similar to the logic in lines 807-811 except for the object the data is finally written to.

I suggest extracting the common logic into a helper method and passing `flash`/`infoFlash` as an argument.